src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -416,46 +416,33 @@ private bool CanUseTupleTypeName(INamedTypeSymbol tupleSymbol)
             return true;
         }
 
-        private static bool HasNonDefaultTupleElementNames(INamedTypeSymbol tupleSymbol)
+        private static bool HasNonDefaultTupleElements(INamedTypeSymbol tupleSymbol)
         {
-            var elementNames = tupleSymbol.TupleElementNames;
-            if (!elementNames.IsDefault)
-            {
-                for (int i = 0; i < elementNames.Length; i++)
-                {
-                    if (elementNames[i] != TupleTypeSymbol.TupleMemberName(i + 1))
-                    {
-                        return true;
-                    }
-                }
-            }
-
-            return false;
+           return tupleSymbol.TupleElements.Any(e=>(object)e != e.CorrespondingTupleField);
```

Comment:
how common is our use of Linq in SymbolDisplay.  I'm slightly concerned about allocations here.  Also, format this line :)